prusti-tests/tests/typecheck/ui/abstract-predicates/invalid-sig-1.stderr

```diff
@@ -0,0 +1,9 @@
+error[E0308]: mismatched types
+  --> $DIR/invalid-sig-1.rs:15:13
+   |
+15 |             42 //~ ERROR: mismatched types
```

Comment:
This still doesn't explain what this error is. Are predicates supposed to only return `bool`? I don't think this should be the case. (If it were, the error here should highlight the return type, not the expression.) I'm guessing this error is raised because of a `(... : bool)` type check expression, which should not exist for predicate spec items (but should for pre/postconditions).

Reply:
I believed that predicates are meant to return `bool`. However, the user docs don't mention that and I think my assumption was wrong after all.
However, this PR doesn't change the encoding/behavior of predicates at all.
I think if we want to fix this we should do it in a separate PR then and I will delete this test.

Reply:
I deleted the test